ldap/client.go

```diff
@@ -525,15 +528,32 @@ func (c *Client) filterGroupsSearch(userDN string, username string) ([]*ldap.Ent
 		return nil, warnings, fmt.Errorf("%s: LDAP search failed due to template parsing error: %w", op, err)
 	}
 
-	result, err := c.conn.Search(&ldap.SearchRequest{
-		BaseDN: c.conf.GroupDN,
-		Scope:  ldap.ScopeWholeSubtree,
-		Filter: renderedQuery.String(),
-		Attributes: []string{
-			c.conf.GroupAttr,
-		},
-		SizeLimit: math.MaxInt32,
-	})
+	var result *ldap.SearchResult
+	switch {
+	case c.conf.MaximumPageSize > 0:
+
+		result, err = c.conn.SearchWithPaging(&ldap.SearchRequest{
```

Comment:
Can we extract the `ldap.SearchRequest` parameter out and reuse it across both `SearchWithPaging` and `Search`? I believe they are the same.

Reply:
ty!